.dagger/cli.go

```diff
@@ -151,14 +202,15 @@ func (cli *CLI) TestPublish(ctx context.Context) error {
 }
 
 func publishEnv(ctx context.Context) (*dagger.Container, error) {
-	// TODO: remove after upgrading to GoReleaser Pro has go 1.23.2 (it currently only has go 1.23.1)
-	go1_23_2 := dag.Container().From("golang:1.23.2-alpine@sha256:9dd2625a1ff2859b8d8b01d8f7822c0f528942fe56cfe7a1e7c38d3b8d72d679").Directory("/usr/local/go")
-
-	ctr := dag.Container().
-		From(fmt.Sprintf("ghcr.io/goreleaser/goreleaser-pro:%s-pro", goReleaserVersion)).
-		WithDirectory("/usr/local/go", go1_23_2).
-		WithEntrypoint([]string{}).
-		WithExec([]string{"apk", "add", "aws-cli"})
+
+	ctr := dag.Container().From(goReleaserImage)
+
+	// HACK: this can be enabled to force a go update (e.g. when we need it for
+	// a security update)
+	// ctr = ctr.WithDirectory(
+	// 	"/usr/local/go",
+	// 	dag.Container().From("golang:<version>-alpine@sha256:<hash>").Directory("/usr/local/go"),
+	// )
```

Comment:
Agreed this is cruft, but it's cruft we'll probably use again - IMO, having this inline for when we inevitably need it next (probably the next go release) is better than having to come back and attempt to reconstruct the logic. I want to try and keep as much tribal knowledge out of the release + maintenance process - the more tribal knowledge we have, the more it rests on just a couple of people.

Actually, the idea is to remove this logic entirely. As part of the release improvements, I want to have goreleaser ingest the completely built binaries, instead of attempting to build them itself - it's frustrating to have the discrepancy we already have (and it's required if we want to persue the one-binary goal). With that, we can entirely remove this snippet.